modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/IsdaCompliantDiscountCurveCalibrator.java

```diff
@@ -0,0 +1,454 @@
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.credit.cds;
+
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.Function;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.date.BusinessDayAdjustment;
+import com.opengamma.strata.basics.date.DayCount;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.array.DoubleMatrix;
+import com.opengamma.strata.data.MarketData;
+import com.opengamma.strata.market.ValueType;
+import com.opengamma.strata.market.curve.ConstantNodalCurve;
+import com.opengamma.strata.market.curve.CurveInfoType;
+import com.opengamma.strata.market.curve.CurveMetadata;
+import com.opengamma.strata.market.curve.CurveName;
+import com.opengamma.strata.market.curve.CurveNode;
+import com.opengamma.strata.market.curve.CurveParameterSize;
+import com.opengamma.strata.market.curve.DefaultCurveMetadata;
+import com.opengamma.strata.market.curve.DepositIsdaCreditCurveNode;
+import com.opengamma.strata.market.curve.InterpolatedNodalCurve;
+import com.opengamma.strata.market.curve.IsdaCreditCurveNode;
+import com.opengamma.strata.market.curve.JacobianCalibrationMatrix;
+import com.opengamma.strata.market.curve.NodalCurve;
+import com.opengamma.strata.market.curve.SwapIsdaCreditCurveNode;
+import com.opengamma.strata.market.curve.interpolator.CurveExtrapolators;
+import com.opengamma.strata.market.curve.interpolator.CurveInterpolators;
+import com.opengamma.strata.market.param.ParameterMetadata;
+import com.opengamma.strata.market.param.UnitParameterSensitivities;
+import com.opengamma.strata.math.impl.matrix.CommonsMatrixAlgebra;
+import com.opengamma.strata.math.impl.matrix.MatrixAlgebra;
+import com.opengamma.strata.math.impl.rootfinding.BracketRoot;
+import com.opengamma.strata.math.impl.rootfinding.NewtonRaphsonSingleRootFinder;
+
+/**
+ * ISDA compliant discount curve calibrator.
+ * <p>
+ * A single discounting curve is calibrated for a specified currency.
+ * <p>
+ * The curve is defined using two or more {@linkplain CurveNode nodes}.
+ * Each node primarily defines enough information to produce a reference trade.
+ * Calibration involves pricing, and re-pricing, these trades to find the best fit using a root finder.
+ * <p>
+ * Once calibrated, the curves are then available for use.
+ * Each node in the curve definition becomes a parameter in the matching output curve.
+ */
+public final class IsdaCompliantDiscountCurveCalibrator {
+
+  /**
+   * Default implementation.
+   */
+  public static final IsdaCompliantDiscountCurveCalibrator DEFAULT = new IsdaCompliantDiscountCurveCalibrator();
```

Comment:
Should copy the design of `CurveCalibrator`. Have a `standard()` static method and a private constructor. Consider whether root finding parameters should be able to be passed in.